graylog2-server/src/main/java/org/graylog/plugins/views/favorites/FavoritesService.java

```diff
@@ -105,4 +110,19 @@ public Optional<FavoritesForUserDTO> create(final FavoritesForUserDTO favorite,
             throw new IllegalStateException("Unable to create a Favorites collection, collection with this id already exists : " + favorite.id());
         }
     }
+
+    @Subscribe
+    public void removeFavoriteOnEntityDeletion(final RecentActivityEvent event) {
+        // if an entity is deleted, we can no longer see it in the lastOpened collection
+        if (event.activityType().equals(ActivityType.DELETE)) {
+            DBObject query = new BasicDBObject();
+            final DBObject modifications = new BasicDBObject("$pull", new BasicDBObject(FavoritesForUserDTO.FIELD_ITEMS, new BasicDBObject(FavoriteDTO.FIELD_ID, event.grn().entity())));
+            db.updateMulti(query, modifications);
+        }
+    }
+
+    @Subscribe
+    public void removeFavoriteEntityOnUserDeletion(final UserDeletedEvent event) {
+        db.remove(DBQuery.is(FavoritesForUserDTO.FIELD_USER_ID, event.userId()));
```

Comment:
"user_id" field may need an index as well, what do you think?

Reply:
:+1: done